server/routes/api/messages.js

```diff
@@ -1,43 +1,68 @@
 const router = require("express").Router();
 const { Conversation, Message } = require("../../db/models");
 const onlineUsers = require("../../onlineUsers");
+const jwt = require("jsonwebtoken");
+const { User } = require("../../db/models");
 
 // expects {recipientId, text, conversationId } in body (conversationId will be null if no conversation exists yet)
 router.post("/", async (req, res, next) => {
   try {
     if (!req.user) {
       return res.sendStatus(401);
     }
+
     const senderId = req.user.id;
     const { recipientId, text, conversationId, sender } = req.body;
 
-    // if we already know conversation id, we can save time and just add it to message and return
-    if (conversationId) {
-      const message = await Message.create({ senderId, text, conversationId });
-      return res.json({ message, sender });
-    }
-    // if we don't have conversation id, find a conversation to make sure it doesn't already exist
-    let conversation = await Conversation.findConversation(
-      senderId,
-      recipientId
-    );
+    const token = req.cookies["JWT"];
+    if (token) {
+      jwt.verify(token, process.env.SESSION_SECRET, (err, decoded) => {
+        if (err) {
+          return next();
+        }
+        User.findOne({
+          where: { id: decoded.id },
+        }).then(async (user) => {
+          if (user.dataValues.id !== sender.id) {
+            res.sendStatus(403);
+          } else {
+            // if we already know conversation id, we can save time and just add it to message and return
+            if (conversationId) {
+              const message = await Message.create({
+                senderId,
+                text,
+                conversationId,
+              });
+              return res.json({ message, sender });
+            }
```

Comment:
The user is still able to send messages to a conversation they aren't involved in. You will need to check the conversation ID to see if the user is in the conversation before sending the message.